cmd/traceflow-plugin/main.go

```diff
@@ -0,0 +1,285 @@
+	"github.com/vmware-tanzu/octant/pkg/view/component"
+	"github.com/vmware-tanzu/octant/pkg/view/flexlayout"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/vmware-tanzu/antrea/pkg/apis/traceflow/v1"
+	clientset "github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned"
+	"github.com/vmware-tanzu/antrea/pkg/graphviz"
+)
+
+var (
+	pluginName                           = "traceflowPlugin"
+	addTfAction                          = "traceflowPlugin/addTf"
+	showGraphAction                      = "traceflowPlugin/showGraphAction"
+	client          *clientset.Clientset = nil
+	kubeConfig                           = "KUBECONFIG"
+)
+
+const (
+	tfNameCol       = "Trace"
+	srcNamespaceCol = "Source Namespace"
+	srcPodCol       = "Source Pod"
+	srcPortCol      = "Source Port"
+	dstNamespaceCol = "Destination Namespace"
+	dstPodCol       = "Destination Pod"
+	dstSvcCol       = "Destination Service"
+	dstPortCol      = "Destination Port"
+	protocolCol     = "Protocol"
+	crdCol          = "Detailed Information"
+	phaseCol        = "Phase"
+)
+
+// This is octant-trace-plugin.
+// The plugin needs to run with octant version v0.10.2 or later.
+func main() {
+	localPlugin := newTraceflowPlugin()
+
+	// Remove the prefix from the go logger since Octant will print logs with timestamps.
+	log.SetPrefix("")
+
+	capabilities := &plugin.Capabilities{
+		ActionNames: []string{addTfAction, showGraphAction},
+		IsModule:    true,
+	}
+
+	// Set up what should happen when Octant calls this plugin.
+	options := []service.PluginOption{
+		service.WithActionHandler(localPlugin.actionHandler),
+		service.WithNavigation(localPlugin.navHandler, localPlugin.initRoutes),
+	}
+
+	p, err := service.Register(pluginName, "a description", capabilities, options...)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Printf("octant-traceflow-plugin is starting")
+	p.Serve()
+}
+
+type traceflowPlugin struct {
+	client *clientset.Clientset
+	graph  string
+}
+
+func newTraceflowPlugin() *traceflowPlugin {
+	config, err := clientcmd.BuildConfigFromFlags("", os.Getenv(kubeConfig))
+	if err != nil {
+		log.Fatalf("Failed to build kubeConfig %v", err)
+	}
+	client, err = clientset.NewForConfig(config)
+	if err != nil {
+		log.Fatalf("Failed to create K8s client for antrea-traceflow-octant-plugin %v", err)
+	}
+	return &traceflowPlugin{
+		client: client,
+		graph:  "",
+	}
+}
+
+func (a *traceflowPlugin) navHandler(request *service.NavigationRequest) (navigation.Navigation, error) {
+	return navigation.Navigation{
+		Title:    "Trace Flow",
+		Path:     request.GeneratePath("components"),
+		IconName: "cloud",
+	}, nil
+}
+
+func (a *traceflowPlugin) actionHandler(request *service.ActionRequest) error {
+	actionName, err := request.Payload.String("action")
+	if err != nil {
+		log.Printf("unable to get input at string: %w", err)
+		return nil
+	}
+
+	switch actionName {
+	case addTfAction:
```

Comment:
We need to update this part of logic.
1. Add the field name and so that we can trace several times without deleting the previous CRD.
2. Support more protocol and our previous hard code is just for function verification.

Reply:
> We need to update this part of logic.
> 
> 1. Add the field name and so that we can trace several times without deleting the previous CRD.
> 2. Support more protocol and our previous hard code is just for function verification.

For the first point, what if the user inputs duplicate traceflow name? As I have not found a good way to inform users like pop-ups, now I just replace the old traceflow with the new one.

Reply:
Yiwei's point makes sense to me. @gran-vmv Do you think we need this extra field? Or maybe we could keep the previous logic by constructing the name by source and destination, then if a duplicate one already exists, we can always delete and recreate a new one?

Reply:
> Yiwei's point makes sense to me. @gran-vmv Do you think we need this extra field? Or maybe we could keep the previous logic by constructing the name by source and destination, then if a duplicate one already exists, we can always delete and recreate a new one?

I think we can use hash code. Add a random string at the end of each traceflow name.

Reply:
Can we use timestamp for this?

Reply:
Good point, then let's try this way.

Reply:
@ZhangYW18 Please also check when Pod name contains dot "."

Reply:
Updated traceflow name to "srcPod-dstPod-YYYYMMDD-HHMMSS".
Regex check for strings is also added.